e2e/scenario_test.go

```diff
@@ -0,0 +1,946 @@
+	t.Run("azurelinuxv2", Scenario_azurelinuxv2)
+	t.Run("azurelinuxv2-airgap", Scenario_azurelinuxv2AirGap)
+	t.Run("azurelinuxv2-arm64", Scenario_azurelinuxv2ARM64)
+	t.Run("azurelinuxv2-arm64-airgap", Scenario_azurelinuxv2ARM64AirGap)
+	t.Run("azurelinuxv2-azurecni", Scenario_azurelinuxv2_azurecni)
+	t.Run("azurelinuxv2-chrony-restarts", Scenario_azurelinuxv2ChronyRestarts)
+	t.Run("azurelinuxv2-custom-sysctls", Scenario_azurelinuxv2CustomSysctls)
+	t.Run("azurelinuxv2-gpu", Scenario_azurelinuxv2gpu)
+	t.Run("azurelinuxv2-gpu-azurecni", Scenario_azurelinuxv2gpu_azurecni)
+	t.Run("azurelinuxv2-wasm", Scenario_azurelinuxv2Wasm)
+	t.Run("marinerv2", Scenario_marinerv2)
+	t.Run("marinerv2-airgap", Scenario_marinerv2AirGap)
+	t.Run("marinerv2-arm64", Scenario_marinerv2ARM64)
+	t.Run("marinerv2-arm64-airgap", Scenario_marinerv2ARM64AirGap)
+	t.Run("marinerv2-azurecni", Scenario_marinerv2_azurecni)
+	t.Run("marinerv2-chrony-restarts", Scenario_marinerv2ChronyRestarts)
+	t.Run("marinerv2-custom-sysctls", Scenario_marinerv2CustomSysctls)
+	t.Run("marinerv2-gpu", Scenario_marinerv2gpu)
+	t.Run("marinerv2-gpu-azurecni", Scenario_marinerv2gpu_azurecni)
+	t.Run("marinerv2-wasm", Scenario_marinerv2Wasm)
+	t.Run("ubuntu1804", Scenario_ubuntu1804)
+	t.Run("ubuntu1804-azurecni", Scenario_ubuntu1804_azurecni)
+	t.Run("ubuntu1804-chrony-restarts", Scenario_ubuntu1804ChronyRestarts)
+	t.Run("ubuntu1804-gpu", Scenario_ubuntu1804gpu)
+	t.Run("ubuntu1804-gpu-azurecni", Scenario_ubuntu1804gpu_azurecni)
+	t.Run("ubuntu2204", Scenario_ubuntu2204)
+	t.Run("ubuntu2204-airgap", Scenario_ubuntu2204AirGap)
+	t.Run("ubuntu2204-arm64", Scenario_ubuntu2204ARM64)
+	t.Run("ubuntu2204-artifact-streaming", Scenario_ubuntu2204ArtifactStreaming)
+	t.Run("ubuntu2204-chrony-restarts", Scenario_ubuntu2204ChronyRestarts)
+	t.Run("ubuntu2204-containerd-url", Scenario_ubuntu2204ContainerdURL)
+	t.Run("ubuntu2204-containerd-version", Scenario_ubuntu2204ContainerdVersion)
+	t.Run("ubuntu2204-custom-ca-trust", Scenario_ubuntu2204CustomCATrust)
+	t.Run("ubuntu2204-custom-sysctls", Scenario_ubuntu2204CustomSysctls)
+	t.Run("ubuntu2204-gpu-a10", Scenario_ubuntu2204gpua10)
+	t.Run("ubuntu2204-gpu-a100", Scenario_ubuntu2204gpua100)
+	t.Run("ubuntu2204-gpu-grid-driver", Scenario_ubuntu2204GPUGridDriver)
+	t.Run("ubuntu2204-gpu-ncv", Scenario_ubuntu2204gpuncv)
+	t.Run("ubuntu2204-gpu-no-driver", Scenario_ubuntu2204gpuNoDriver)
+	t.Run("ubuntu2204-private-kube-pkg", Scenario_ubuntu2204privatekubepkg)
+	t.Run("ubuntu2204-wasm", Scenario_ubuntu2204Wasm)
+}
+
+func Scenario_azurelinuxv2(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "Tests that a node using a AzureLinuxV2 (CgroupV2) VHD can be properly bootstrapped",
+		Config: Config{
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDAzureLinuxV2Gen2,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-azurelinux-v2-gen2"
```

Comment:
it would be nice to create a shared library of helpers that perform common sorts of mutations on the bootstrap config, e.g. `SetDistro`, I've been meaning to do that for a while but maybe that's something you can take care of

Reply:
There is a limited duplication in a close proximity, so it's not too bad.

Helpers I can think about just complicates things. Maybe `VMSSSKU` can be an extra scenario parameter. Idk.

Reply:
fair... I just think it would help protect against instances in the future where test writers want to set a distro and e.g. only set it on `AgentPoolProfiles[0]`, but mistakenly don't also set it on `Properties.Distro` as well, leading to unexpected behavior

Reply:
in reality we shouldn't have a slice of agent pool's in the NBC at all :)

there was a separate effort at one point to completely remove that aspect from the contract, not sure what ended up happening with that..